eventlet/support/greendns.py

```diff
@@ -179,11 +179,16 @@ def _readlines(self):
         Note that this performs disk I/O so can be blocking.
         """
         lines = []
+        if sys.version_info < (3, 4):
+            mode = 'rU'
+        else:
+            mode = 'r'
```

Comment:
Why use `rb`:

https://docs.python.org/2/library/functions.html#open

> ... The default is to use text mode, which may convert '\n' characters to a platform-specific representation on writing and back on reading. Thus, when opening a binary file, you should append 'b' to the mode value to open the file in binary mode, which will improve portability. (Appending 'b' is useful even on systems that don’t treat binary and text files differently, where it serves as documentation.)

AFAIK, the only platform with explicit support for "text files" nonsense is Windows.

I think it would be more efficient for computing resource usage and for code clarity to just read whole file in one big read and iterate lines broken by any newline-like character. Which is what universal lines do, but without unnecessary substitution of line ending characters.